crates/sql-sink/src/db.rs

```diff
@@ -1,23 +1,242 @@
 use anyhow::anyhow;
```

Comment:
Why there was a need for such a massive rewrite? You could just introduce new trait `Upsert<DB: Database> :Insert` and impl it for SQLite and Postgres where you define a new `query` method. It would take 100 new lines max.

Reply:
I was trying to use `AnyConnection` first to remove generics and repeated code and I tried bunch of other approaches and in the end it evolved into this. I realize in hindsight that it would probably be easier to do what you describe

Reply:
Yeah, `AnyConnection` looks tempting, I also tried to do that but didn't succeed.

Generics are meant to reduce repeated code. No need to remove them.  I spent a lot of thinking to come up with such a solution. And the new one is less compact and more verbose, and I don't see any advantages.